### PR TITLE
Single threaded comments

### PR DESCRIPTION
**Pull request by IanPhilips**
This allows users to reply to comments and answers (previously they commented on answers). When users first hit reply to a comment or thread they're smoothly scrolled to the input and a tag is added to the input if they're not replying to their own comment. 

Hitting enter on large screens submits the comment (shift + enter adds a newline). For smaller screens a Comment or Reply button is shown. 

Overall commenting now replicates facebook's interaction style.

### CODE REVIEW

**Comment by IanPhilips** [comment]
Also note: replies to comments are oldest first whereas general comments are oldest last. Not sure how we want to do this but went with James' suggestion in the notion which feels pretty natural.

https://www.notion.so/manifoldmarkets/Comments-46d9337debd3425281303b3d509b9e86#9b39e38b34d64d3d9f330436c697ba0d

**Comment by IanPhilips** [comment]
Also note: replies do not yet work when in feed mode, just contract view mode. This should be easy to extend, however.

File: web/components/feed/feed-items.tsx [hunk 1]
```diff
@@ -118,24 +118,80 @@ function FeedItem(props: { item: ActivityItem }) {
...
+    router.replace(scrollToId, undefined, { shallow: true })
+    setShowCommentReplyInput(true)
+  }
+  return (
+    <div className={'w-full flex-col flex-col space-x-8 pr-6'}>
+      {comments.map((comment, commentIdx) => (
+        <div
+          key={comment.id}
+          id={comment.id}
+          className={clsx('flex space-x-3', commentIdx === 0 ? '' : 'mt-4')}
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:150]: Maybe add ml-8 and remove the space-x-8 above to be clearer how the indentation works

File: web/components/feed/feed-items.tsx [hunk 2]
```diff
@@ -118,24 +118,80 @@ function FeedItem(props: { item: ActivityItem }) {
...
+        >
+          <FeedComment
+            contract={contract}
+            comment={comment}
+            betsBySameUser={betsBySameUser}
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:155]: Maybe commentIdx === 0 ? betsBySameUser : []

Or, pass all user bets to FeedCommentThread and pass down bets by creator of this comment.

File: web/components/feed/feed-items.tsx [hunk 3]
```diff
@@ -118,24 +118,80 @@ function FeedItem(props: { item: ActivityItem }) {
...
+          />
+        </div>
+      ))}
+      {showCommentReplyInput && (
+        <div className={' w-full pt-6'}>
+          <CommentInput
+            contract={contract}
+            // Should we allow replies to contain recent bet info?
+            betsByCurrentUser={[]}
+            comments={[]}
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:168]: Pass these down, or instead show no bet info for replies

File: web/components/feed/feed-items.tsx [hunk 4]
```diff
@@ -768,14 +872,17 @@ function FeedAnswerGroup(props: {
...
 }) {
-  const { answer, items, contract, type } = props
+  const { answer, items, contract, type, betsByCurrentUser, comments } = props
   const { username, avatarUrl, name, text } = answer
 
   const prob = getDpmOutcomeProbability(contract.totalShares, answer.id)
   const probPercent = formatPercent(prob)
   const [open, setOpen] = useState(false)
-
+  const [showReply, setShowReply] = useState(false)
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:884]: If there's a recent bet by current user, showReply

File: web/components/feed/feed-items.tsx [hunk 5]
```diff
@@ -808,27 +915,71 @@ function FeedAnswerGroup(props: {
...
-              <BuyButton
-                className={clsx(
-                  'btn-sm flex-initial !px-6 sm:flex',
-                  tradingAllowed(contract) ? '' : '!hidden'
-                )}
-                onClick={() => setOpen(true)}
-              />
+            <Row className="items-center justify-center gap-4">
+              {isFreeResponseContractPage && (
+                <div className={'sm:hidden md:hidden lg:hidden'}>
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:920]: sm:hidden is sufficient: it applies to all larger sizes

File: web/components/feed/feed-items.tsx [hunk 6]
```diff
@@ -118,24 +118,80 @@ function FeedItem(props: { item: ActivityItem }) {
...
+  betsBySameUser: Bet[]
+  truncate?: boolean
+  smallAvatar?: boolean
+}) {
+  const { contract, comments, betsBySameUser, truncate, smallAvatar } = props
+  const [showCommentReplyInput, setShowCommentReplyInput] = useState(false)
+  const [replyToUsername, setReplyToUsername] = useState('')
+  const router = useRouter()
+  const scrollToId = '#' + comments[comments.length - 1].id
+  function scrollAndOpenReplyInput(comment: Comment) {
```
Comment (jahooma) [inline web/components/feed/feed-items.tsx:138]: Use useState to set the ref to the textarea HTMLTextAreaElement, and set focus here

**Review by jahooma — COMMENTED** [review]
Awesome! This will be a hit I'm sure!

**Pull request merged**